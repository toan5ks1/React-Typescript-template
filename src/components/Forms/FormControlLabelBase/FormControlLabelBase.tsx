import { FormControlLabel, FormControlLabelProps } from "@mui/material";

export interface FormControlLabelBaseProps extends FormControlLabelProps {
    disableMarginLeft?: boolean;
    disableMarginRight?: boolean;
}

const FormControlLabelBase = (props: FormControlLabelBaseProps) => {
    const { disableMarginLeft, disableMarginRight, sx = [], ...rest } = props;

    return (
        <FormControlLabel
            {...rest}
            sx={[
                disableMarginLeft ? { marginLeft: 0 } : {},
                disableMarginRight ? { marginRight: 0 } : {},
                ...(Array.isArray(sx) ? sx : [sx]),
            ]}
        />
    );
};

export default FormControlLabelBase;
