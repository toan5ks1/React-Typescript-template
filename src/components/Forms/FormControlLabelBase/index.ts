export { default } from "./FormControlLabelBase";
export * from "./FormControlLabelBase";
