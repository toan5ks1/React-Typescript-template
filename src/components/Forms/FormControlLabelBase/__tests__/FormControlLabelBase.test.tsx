import FormControlLabelBase from "../FormControlLabelBase";

import { render } from "@testing-library/react";

const defaultFormControlLabelBaseProps = {
    name: "switchLabel",
    label: "Switch Label",
    disableMarginLeft: false,
    disableMarginRight: false,
    control: <input />,
};

describe(FormControlLabelBase.name, () => {
    it("should match snapshot", () => {
        const { container } = render(<FormControlLabelBase control={<input />} label="label" />);
        expect(container).toMatchSnapshot();
    });
});

describe("<FormControlLabelBase />", () => {
    it("render component with disableMarginRight should have margin-right = 0px", () => {
        const wrapper = render(
            <FormControlLabelBase
                {...defaultFormControlLabelBaseProps}
                disableMarginRight
                data-testid="FormControlLabelBase"
            />
        );

        expect(wrapper.getByTestId("FormControlLabelBase")).toHaveStyle("margin-right: 0px");
    });

    it("render component with disableMarginLeft should have margin-left = 0px", () => {
        const wrapper = render(
            <FormControlLabelBase
                {...defaultFormControlLabelBaseProps}
                disableMarginLeft
                data-testid="FormControlLabelBase"
            />
        );

        expect(wrapper.getByTestId("FormControlLabelBase")).toHaveStyle("margin-left: 0px");
    });

    it("render component with disableMarginLeft & disableMarginRight should have margin-left = 0px & margin-right: 0px", () => {
        const wrapper = render(
            <FormControlLabelBase
                {...defaultFormControlLabelBaseProps}
                disableMarginLeft
                disableMarginRight
                sx={{ color: "red" }}
                data-testid="FormControlLabelBase"
            />
        );

        expect(wrapper.getByTestId("FormControlLabelBase")).toHaveStyle(
            `margin-left: 0px; margin-right: 0px; color: red;`
        );
    });
});
