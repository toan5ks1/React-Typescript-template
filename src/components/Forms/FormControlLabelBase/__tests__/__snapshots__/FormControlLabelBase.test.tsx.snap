// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControlLabelBase should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.emotion-0.Mui-disabled {
  cursor: default;
}

.emotion-0 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-0"
  >
    <input />
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-1"
    >
      label
    </span>
  </label>
</div>
`;
